src/overlays/actors/ovl_En_Water_Effect/z_en_water_effect.c

```diff
@@ -28,30 +46,672 @@ const ActorInit En_Water_Effect_InitVars = {
+            Matrix_InsertYRotation_f(ptr->unk_2C.z, MTXMODE_APPLY);
+
+            gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+            gSPDisplayList(POLY_XLU_DISP++, D_060042F8);
+        }
+    }
+
+    CLOSE_DISPS(gfxCtx);
+}
+
+void func_80A599E8(EnWaterEffect* this, Vec3f* arg1, u8 arg2) {
+    s16 i;
+    EnWaterEffectStruct* ptr = &this->unk_144[0];
+
+    for (i = 0; i < ARRAY_COUNT(this->unk_144); i++, ptr++) {
+        if (ptr->unk_00 == 0) {
+            ptr->unk_00 = 4;
+            ptr->unk_04 = *arg1;
+
+            ptr->unk_1C = D_80A5AFB0;
+            ptr->unk_10 = D_80A5AFB0;
+
+            if ((arg2 == 0) || (arg2 == 2)) {
+                ptr->unk_1C.y = -1.0f;
+            }
+
+            if (arg2 >= 2) {
+                if (arg2 == 2) {
+                    ptr->unk_10.x = randPlusMinusPoint5Scaled(10.0f);
+                    ptr->unk_10.y = Rand_ZeroFloat(3.0f) + 5.0f;
+                    ptr->unk_10.z = randPlusMinusPoint5Scaled(10.0f);
+                }
+                ptr->unk_2C.z = 0.0017f;
+                ptr->unk_2C.x = 0.003f;
+                ptr->unk_2C.y = 0.0018f;
+            } else {
+                ptr->unk_2C.z = 0.003f;
+                ptr->unk_2C.x = 0.003f;
+                ptr->unk_2C.y = 0.003f;
+            }
+            ptr->unk_38 = 255.0f;
+            ptr->unk_28 = Rand_ZeroFloat(0x10000);
+            ptr->unk_3C = 255;
+            ptr->unk_01 = Rand_ZeroFloat(200.0f);
+            ptr->unk_2A = arg2;
+            break;
+        }
+    }
+}
+
+void func_80A59C04(Actor* thisx, GlobalContext* globalCtx2) {
+    GlobalContext* globalCtx = globalCtx2;
+    EnWaterEffect* this = THIS;
+    s16 i;
+    s16 j;
+    f32 temp_f0_2;
+    Player* player = GET_PLAYER(globalCtx);
+    Vec3f sp90;
+    Actor* rotaryRoom = globalCtx->actorCtx.actorList[ACTORCAT_BG].first;
+    CollisionPoly* sp88;
+    EnWaterEffectStruct* ptr = &this->unk_144[0];
+    u8 phi_s5;
+    Vec3f sp74;
+
+    while (rotaryRoom != NULL) {
+        if (rotaryRoom->id == ACTOR_BG_IKANA_ROTARYROOM) {
+            break;
+        }
+        rotaryRoom = rotaryRoom->next;
+    }
+
+    if (this->unk_DC6 != 0) {
+        this->unk_DC6--;
+    }
+
+    if ((rotaryRoom != NULL) && Flags_GetSwitch(globalCtx, (rotaryRoom->params >> 1) & 0x7F)) {
+        this->unk_DC6 = Rand_ZeroFloat(150.0f) + 100.0f;
+    } else if (!func_801690CC(globalCtx)) {
+        this->unk_DC4++;
+        if (this->unk_DC6 == 0) {
+            this->unk_DC6 = Rand_ZeroFloat(150.0f) + 100.0f;
+            Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 20.0f, 10.0f, 40.0f, 4);
+            sp74.x = randPlusMinusPoint5Scaled(50.0f) + this->actor.world.pos.x;
+            sp74.y = this->actor.world.pos.y;
+            sp74.z = randPlusMinusPoint5Scaled(50.0f) + this->actor.world.pos.z;
+            func_80A599E8(this, &sp74, 0U);
```

Comment:
```suggestion
            func_80A599E8(this, &sp74, 0);
```
Needed? Could you also mess around with the prototype of `func_80A599E8`? Applies to other cases of `func_80A599E8` below